### PR TITLE
Move some queries from rustc::ty to librustc_ty.

### PR DESCRIPTION
**Pull request by cjgillot**
cc #65031

### CODE REVIEW

**Comment by Zoxc** [comment]
I think it would make sense to move this to a new `rustc_ty` crate. cc @rust-lang/compiler

File: src/librustc_passes/ty.rs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+//pub fn is_impl_trait_defn(tcx: TyCtxt<'_>, def_id: DefId) -> Option<DefId> {
+//    if let Some(hir_id) = tcx.hir().as_local_hir_id(def_id) {
+//        if let Node::Item(item) = tcx.hir().get(hir_id) {
+//            if let hir::ItemKind::OpaqueTy(ref opaque_ty) = item.kind {
+//                return opaque_ty.impl_trait_fn;
+//            }
+//        }
+//    }
+//    None
+//}
```
Comment (estebank) [inline src/librustc_passes/ty.rs]: ?

**Comment by matthewjasper** [comment]
Nominating for discussion of `rustc_ty`

File: src/librustc_ty/ty.rs [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    let node = tcx.hir().get(hir_id);
+
+    let fn_like = hir_map::blocks::FnLikeNode::from_node(node).unwrap_or_else(|| {
+        bug!("asyncness: expected fn-like node but got `{:?}`", def_id);
+    });
+
+    fn_like.asyncness()
+}
+
+pub fn provide(providers: &mut ty::query::Providers<'_>) {
```
Comment (Zoxc) [inline src/librustc_ty/ty.rs]: I guess you need to call this.

Comment (cjgillot) [reply src/librustc_ty/ty.rs]: Oops, sorry. Added a commit for that.

**Pull request merged**